taqo_cli/bin/taqo_cli.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:taqo_cli/platform_service.dart';
+import 'package:taqo_cli/taqo_cli.dart' as taqo_cli;
+import 'package:taqo_common/service/logging_service.dart';
+import 'package:taqo_common/storage/base_database.dart';
+
+Future<void> main(List<String> arguments) async {
+  if (arguments.length < 1) {
+    displayUsageAndExit(1);
+  }
+
+  // await LoggingService.initialize(
+  //     logFilePrefix: 'cli-', outputsToStdout: false);
+  DatabaseFactory.initialize(() => databaseImpl);
+  final taqoCli = taqo_cli.TaqoCli();
+
+  switch (arguments[0]) {
+    case 'join':
+      if (arguments.length != 2) {
+        displayUsageAndExit(1);
+      }
+      await taqoCli.joinPublicExperimentWithInvitationCode(arguments[1]);
```

Comment:
What happens if they try a second invitation code on the same experiment?

Reply:
The first join will become invalid (overwritten) and they will join again with a different participant id